src/authorize.ts

```diff
@@ -56,7 +65,15 @@ export interface IEveryAuthOptions {
    * subsequent requests.  Usually extracts the value from the output of the authorization system, or can be a
    * constant for experimentation and testing purposes.
    */
-  mapToUserId: ((req: express.Request) => Promise<string>) | ((req: express.Request) => string);
+  mapToUserId?: ((req: express.Request) => Promise<string>) | ((req: express.Request) => string);
+
+  /**
+   * Called after a user successfully authorized to a target service but before their identity has been persisted.
+   * Perform any side-effect operations like removing prior identities that are no longer needed.
+   */
+  onAuthorized?:
+    | ((req: express.Request, ctx: IAuthorizedContext) => Promise<void>)
+    | ((req: express.Request, ctx: IAuthorizedContext) => void);
```

Comment:
Should the other types also be adjusted to be optionally-async?  Ditto for docs.

Reply:
Which other APIs? I modeled this one on what mapToUserId and mapToTenantId were doing.